crates/swc_ecma_transforms_compat/src/es2015/block_scoping/operator.rs

```diff
@@ -26,6 +26,26 @@ impl VisitMut for Rename {
         }
     }
 
+    fn visit_mut_prop(&mut self, n: &mut Prop) {
+        let span = n.span();
+        if span.ctxt == SyntaxContext::empty() {
```

Comment:
Should I omit this check logic?

Reply:
I think this logic can cause a problem

Reply:
Why did you add this logic?

Reply:
To skip prop which generated by ourselves.
I noticed that it's the reason caused the test result change.
I will correct it.

https://github.com/swc-project/swc/blob/6a05c20a19d048031cc25e20dedbc517a31cc346/crates/swc/tests/tsc-references/intraExpressionInferences.1.normal.js#L91-L93

https://github.com/swc-project/swc/blob/6a05c20a19d048031cc25e20dedbc517a31cc346/crates/swc_ecma_parser/tests/tsc/intraExpressionInferences.ts#L98-L100